### DISCUSSION TITLE
banning display names after the new discord patch

### DISCUSSION THREAD

**Discussion by lmaotrix** [Rapptz/discord.py#9481] [Q&A]
```
from discord.ext import commands
from config import USER_ID


banned_names = [
    'kiyo',
    'taka',
    'ayano',
    'koji',
    'ayanokoji',
    'ayanokouji',
    'sainlow',
    'pon',
    'aya'
]

class BanUsers(commands.Cog):
    def init(self, bot):
        self.bot = bot

    @commands.command()
    async def banusers(self, ctx):
        if ctx.author.id == USER_ID:
            print('executing')
            print(ctx.guild.me.guild_permissions.ban_members)
            banned_users = []
            for member in ctx.guild.members:
                if any(name.lower() in member.display_name.lower() for name in banned_names):
                    await member.ban(reason='ayanokoji wannabe issue')
                    banned_users.append(member.display_name)
            if banned_users:
                await ctx.send('All ayanokoji wannabes have been banned')
            else:
                await ctx.send('Server is clear!')
        else:
            await ctx.send(f"{ctx.author.mention}, you're not worthy of using me.")
        pass

async def setup(bot):
    await bot.add_cog(BanUsers(bot))
```

basically this is the code for a command that should ban all the members of a server with display name that contains whatever's in the dictionary on the top, but for some reason it just responds with 'server is clear' even when the members with those certain display names are actually there
anyone who can help?

here is the link to the github repo in case you wanted to give it a further look: https://github.com/lmaotrix/discord-purgebot

**Comment by Soheab** [comment]
The code you linked seems to be checking the username over display_name:
https://github.com/lmaotrix/discord-purgebot/blob/cca52ac05521ed11df6b6fe17e9a09a7571a5c2f/bot/src/cogs/banusers.py#L29


FYI, `Member.display_name` is the nickname or global_name or username, `.global_name` would be the actual "display name" as it's called in the UI.

Also, consider using the [server](https://discord.com/invite/dpy) for any further help :)

**Reply by lmaotrix** [reply]
Thank you very much, didn't know global_user was a thing hahah. Couldn't find this on the documentation too, might've missed it

**Comment by lmaotrix** [comment]
> The code you linked seems to be checking the username over display_name:
> 
> https://github.com/lmaotrix/discord-purgebot/blob/cca52ac05521ed11df6b6fe17e9a09a7571a5c2f/bot/src/cogs/banusers.py#L29
> 
> 
> 
> 
> 
> FYI, `Member.display_name` is the nickname or global_name or username, `.global_name` would be the actual "display name" as it's called in the UI.
> 
> 
> 
> Also, consider using the [server](https://discord.com/invite/dpy) for any further help :)


Do you mind if I ask what the difference between member.display_name and member.name?

**Reply by Soheab** [reply]
[Member.display_name](https://discordpy.readthedocs.io/en/stable/api.html#discord.Member.display_name) basically does `return self.nick or self.global_name or self.name`.

[Member.name](https://discordpy.readthedocs.io/en/stable/api.html#discord.Member.name) is the unique username (`self.name`).
